### PR TITLE
Collapse comment by thread border

### PR DESCRIPTION
**Pull request by akellbl4**
#162

### CODE REVIEW

**Comment by umputun** [comment]
thx. Not sure if expected, but the selection seems to work for the single (current?) segment only, i.e.:

image

Another thing - there is no visual indication of a collapsed tree

**Comment by akellbl4** [comment]
@umputun, ok. I realize what I need to change.

**Comment by umputun** [comment]
Not sure if you expected me to review, but if you do - I still don't understand what it's doing. It seems to collapse a part of the level sometime (also indicates incomplete level), and sometimes it collapses a level above. Just to make sure we are on the same page - this thing supposes to do exactly the same fold as the current [-] icon does.

Indication of minimized subtree odd as well (losing avatar and a part of the name). I'd suggest something simple here, like show the "title" in italic or different shade or smth like this

**Comment by akellbl4** [comment]
All right. I made the right logic of comment collapsing and fixed collapse on the deepest level.
Now I am going to "make it sexy" and to do this code clean.

File: frontend/app/components/thread/thread.tsx [hunk 1]
```diff
@@ -29,10 +21,21 @@ interface OwnProps {
...
+  const collapsed = getThreadIsCollapsed(comment)(state);
+
+  return { comment, childs, collapsed, theme };
+};
 
-function Thread(props: RenderableProps<Props>) {
-  const { collapsed, comment, childs, level, theme } = props;
+export const Thread: FunctionComponent<Props> = ({ id, level, mix, getPreview }) => {
+  const dispatch = useDispatch();
+  const { collapsed, comment, childs, theme } = useSelector(commentSelector(id));
```
Comment (Mavrin) [inline frontend/app/components/thread/thread.tsx]: Please use shallow equality https://react-redux.js.org/next/api/hooks#equality-comparisons-and-updates
 or use separate useSelector

**Comment by umputun** [comment]
>  I would expect a rectangle or plus sign or something else which will tell me that it's something I can expand.

We have tried some other icons but found nothing decent. Open for suggestions/examples

> With the current design, I would expect that clicking an avatar would collapse the thread: it's not the case now.

Well, this functionality already occupied by the user's widget. I don't think changing the behavior of such a click based on view is a good idea.

> The deepest level of comments (or I think deeper than the deepest level) is noncollapsible.

They seem to be collapsable from the closest top level with a visual ident. I don't want to make it complicated and allow some smart collapses for deepest level. To me, it should match visual borders and if level 10 collapsed because user clicked on the deepest visual ident border (6?) - so be it.

**Comment by paskal** [comment]
> We have tried some other icons but found nothing decent. Open for suggestions/examples

I don't think we should preserve user avatar, that's a confusing part. Even having the current icon without an avatar background would be better in my opinion.

>> With the current design, I would expect that clicking an avatar would collapse the thread: it's not the case now.

> Well, this functionality already occupied by the user's widget. I don't think changing the behavior of such a click based on view is a good idea.

Clicking avatar doesn't produce any action in the current version of remark, as far as I know. I am OK with keeping the current behavior, however.

> They seem to be collapsable from the closest top level with a visual ident. I don't want to make it complicated and allow some smart collapses for deepest level. To me, it should match visual borders and if level 10 collapsed because user clicked on the deepest visual ident border (6?) - so be it.

Good by me.

**Comment by umputun** [comment]
> I don't think we should preserve user avatar, that's a confusing part. Even having the current icon without an avatar background would be better in my opinion.

interesting idea, worth to try

> Clicking avatar doesn't produce any action in the current version of remark, as far as I know.

yeah, you right. Probably click on avatar can do collapse

**Comment by akellbl4** [comment]
> Probably click on avatar can do collapse

I think collapsing by click on avatar might be confusing because it is not a common behavior for avatar.

**Comment by umputun** [comment]
>  by click on avatar might be confusing 

agree

**Pull request merged**